src/Accordion.js

```diff
@@ -0,0 +1,99 @@
+import React, { cloneElement, useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const Accordion = ({
+    children,
+    className,
+    closedClassName,
+    defaultActiveItem,
+    icon,
+    iconClassName,
+    onClosing,
+    onOpening,
+    onToggle,
+    multiple,
+    scrollIntoView,
+}) => {
+    const [activeIndex, setActiveIndex] = useState(() => {
+        switch (true) {
+        case defaultActiveItem != null:
+            if (multiple) {
+                return Array.isArray(defaultActiveItem) ? defaultActiveItem : [defaultActiveItem];
+            }
+
+            return Array.isArray(defaultActiveItem) ? defaultActiveItem[0] ?? null : defaultActiveItem;
+
+        default:
+            return multiple ? [] : null;
+        }
+    });
+
+    const setOpenItem = useCallback((index) => {
+        switch (true) {
+        case multiple:
+            if (activeIndex.includes(index)) {
+                onClosing && onClosing(index);
+                setActiveIndex(activeIndex.filter((i) => i !== index));
+            } else {
+                onOpening && onOpening(index);
```

Comment:
I would have the onOpening (onOpen) prop with a default noop, so that you don't have to check this every time and you can just simply call it.